src/controller/java/AndroidCallbacks.cpp

```diff
@@ -907,6 +915,77 @@ void InvokeCallback::ReportError(const char * message, ChipError::StorageType er
+                            ChipLogProgress(Controller, "Get TLV Value fail!"));
+        err = JniReferences::GetInstance().CreateBoxedObject<jlong>("java/lang/Long", "(J)V", static_cast<jlong>(unsignedValue),
+                                                                    retValue);
+        VerifyOrReturnValue(err == CHIP_NO_ERROR, nullptr, ChipLogProgress(Controller, "Create Boxed Object fail!"));
+        return retValue;
+    }
+    case TLV::kTLVType_Boolean: {
+        bool booleanValue;
+        VerifyOrReturnValue(readerForJavaObject.Get(booleanValue) == CHIP_NO_ERROR, nullptr,
+                            ChipLogProgress(Controller, "Get TLV Value fail!"));
+        err = JniReferences::GetInstance().CreateBoxedObject<jboolean>("java/lang/Boolean", "(Z)V",
+                                                                       static_cast<jboolean>(booleanValue), retValue);
+        VerifyOrReturnValue(err == CHIP_NO_ERROR, nullptr, ChipLogProgress(Controller, "Create Boxed Object fail!"));
+        return retValue;
+    }
+    case TLV::kTLVType_FloatingPointNumber: {
+        double doubleValue;
+        VerifyOrReturnValue(readerForJavaObject.Get(doubleValue) == CHIP_NO_ERROR, nullptr,
+                            ChipLogProgress(Controller, "Get TLV Value fail!"));
+        err = JniReferences::GetInstance().CreateBoxedObject<jdouble>("java/lang/Double", "(D)V", static_cast<jdouble>(doubleValue),
+                                                                      retValue);
+        VerifyOrReturnValue(err == CHIP_NO_ERROR, nullptr, ChipLogProgress(Controller, "Create Boxed Object fail!"));
+        return retValue;
+    }
+    case TLV::kTLVType_UTF8String: {
+        uint32_t bufferLen             = readerForJavaObject.GetLength();
+        std::unique_ptr<char[]> buffer = std::unique_ptr<char[]>(new char[bufferLen + 1]);
+        err                            = readerForJavaObject.GetString(buffer.get(), bufferLen + 1);
+        VerifyOrReturnValue(err == CHIP_NO_ERROR, nullptr, ChipLogProgress(Controller, "Get TLV Value fail!"));
+        chip::CharSpan valueSpan(buffer.get(), bufferLen);
+        chip::JniReferences::GetInstance().CharToStringUTF(valueSpan, retValue);
+        return retValue;
+    }
+    case TLV::kTLVType_ByteString: {
+        uint32_t bufferLen                = readerForJavaObject.GetLength();
+        std::unique_ptr<uint8_t[]> buffer = std::unique_ptr<uint8_t[]>(new uint8_t[bufferLen + 1]);
+        err                               = readerForJavaObject.GetBytes(buffer.get(), bufferLen + 1);
+        VerifyOrReturnValue(err == CHIP_NO_ERROR, nullptr, ChipLogProgress(Controller, "Get TLV Value fail!"));
+
+        jbyteArray valueByteArray = env->NewByteArray(static_cast<jsize>(bufferLen));
+        env->SetByteArrayRegion(valueByteArray, 0, static_cast<jsize>(bufferLen), reinterpret_cast<const jbyte *>(buffer.get()));
+
+        return static_cast<jobject>(valueByteArray);
+    }
+    default:
```

Comment:
should we handle struct and array?, and dump it into json?

Reply:
It seems we return pure objects here ... it sounds like list/array should be a list as well. Unsure about structures though.

Reply:
@yunhanw-google 
The purpose of this modification is to transmit at least some values of the Custom Attribute through TLV Parsing. As andy31415 commented, it may be possible to implement an Array/List with a simple value, but I think it will be difficult to implement anything else.

Reply:
thanks!